hashes/README.md

```diff
@@ -30,3 +30,90 @@ Alternatively add symlinks in your `.git/hooks` directory to any of the githooks
 
 We use a custom Rust compiler configuration conditional to guard the bench mark code. To run the
 bench marks use: `RUSTFLAGS='--cfg=bench' cargo +nightly bench`.
+
+## Algorithms
+
+We currently support the following hash algorithms
+
+### [SHA-1](https://en.wikipedia.org/wiki/SHA-1)
+
+Since 2005, SHA-1 has not been considered secure against well-funded opponents
```

Comment:
In ce1881bbfd8029cce25799ba0668ebae43259b9b:

I would either drop this line or qualify "secure" and "well-funded" much more precisely. This is quite a provocative line when applied to a hash used by git (2005) and bitcoin (2009)! (I checked the wikipedia page which has these exact words, so I guess I don't blame you....but wtf. They link to a blog post by Bruce Schneier which is slightly less provocative, which in turn cites an actual result, which is that SHA1 had 2^69 security against collisions in 2005.)

Since 2017 (SHAttered) sha1 has not been collision resistant even against poorly-funded components. I would suggest instead writing:

```
SHA-1 has seen a series of cryptographic breaks and should not be considered collision-resistant. While there are no practical weaknesses against its first- or second-preimage security, we do not recommend it for any modern application.
```
and we could link "should not be considered collision-resistant" to the wiki page or the SHAttered paper or press release or whatever. Or to the bitcoin transaction which claimed petertodd's "collide sha1 bounty" transaction :).